explainer.md

```diff
@@ -0,0 +1,255 @@
+      border: 10px solid black;
+      border-radius: 10% 10%;
+      box-shadow: 0px 0px 10px;
+    }
+    
+    ::shared-old(#shared-header) {
+      top: -10px;
+      left: -10px;
+    }
+  </style>
+</head>
+<body>
+  <div class="animated" shared-id="shared-header">hello world</div>
+</body>
+</html>
+```
+
+This capability requires the following changes to the steps mentioned in the design above :
+* A new pseudo class ("transition") is introduced which is activated when an element is being displayed using pseudo elements in step 1 and 5.
+* When saving state in step 3, the container element's complete style is preserved instead of only the computed size and transform.
+
+## Modifications to element()
+The following changes will be made to the element() spec as a part of this proposal :
+
+* The target element must have paint containment (contain:paint) to ensure the element is the containing block for all positioned descendents and generates a stacking context.
+* The target element must disallow fragmentation (break-inside:avoid).
+* A new cached-element() function is introduced to refer to the saved output of the element() function in step 3) of the design.
+
+# API Examples
+## SPA
+TODO: Add example
+
+## MPA Script Based
+TODO: Add example
+
+# Alternatives Considered
+## Heirarchical Properties
+This proposal disallows a shared element to be nested inside another shared element. The restriction avoids the need to preserve the hierarchy of the shared elements and associated properties (transform, clip, effects inherited by descendents) when creating pseudo elements. This is a consideration for future iterations of the feature.
+
+## Container/Child Split
+One consideration was to render each shared element using a replaced element directly instead of creating a container element. The motivation behind this split is to provide a stacking context to cross-fade the content of old and new shared elements. This was necessary to ensure blending identical pixels is a no-op using [plus-lighter](https://drafts.fxtf.org/compositing/#porterduffcompositingoperators_plus_lighter) blending. While same-origin transitions could avoid this, it enables future extensibility for cross-origin transitions where cross-fading identical images would be common.
+
+## Natively Supported Live Animatable Properties
+An alternate approach to the setup described in [Live Animatable Properties](#live-animatable-properties) is to support this natively in the browser by introducing a new content-element() function. This function would behave similar to the element() function except skipping the following properties when painting the element: box decorations and visual effects which generate a stacking context. The image will also be sized to the element's content-box (as opposed to the border-box used by the element() function). The motivation for supporting this natively would be to make these properties animatable instead of requiring developer side changes.
```

Comment:
```suggestion
An alternate approach to the setup described in [Animating Box Decoration CSS Properties](#live-animatable-properties) is to support this natively in the browser by introducing a new content-element() function. This function would behave similar to the element() function except skipping the following properties when painting the element: box decorations and visual effects which generate a stacking context. The image will also be sized to the element's content-box (as opposed to the border-box used by the element() function). The motivation for supporting this natively would be to make these properties animatable instead of requiring developer side changes.
```

Reply:
Done.